src/plugins/intel_gpu/src/graph/impls/ocl/experimental_detectron_generate_proposals_single_image.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "experimental_detectron_generate_proposals_single_image_inst.hpp"
+#include "primitive_base.hpp"
+#include "impls/implementation_map.hpp"
+#include "kernel_selector_helper.h"
+#include "edgpsi/experimental_detectron_generate_proposals_single_image_kernel_selector.h"
+#include "edgpsi/experimental_detectron_generate_proposals_single_image_kernel_ref.h"
+
+
+namespace cldnn {
+namespace ocl {
+struct experimental_detectron_generate_proposals_single_image_impl
+        : public typed_primitive_impl_ocl<experimental_detectron_generate_proposals_single_image> {
+    using parent = typed_primitive_impl_ocl<experimental_detectron_generate_proposals_single_image>;
+    using parent::parent;
+
+    std::unique_ptr<primitive_impl> clone() const override {
+        return make_unique<experimental_detectron_generate_proposals_single_image_impl>(*this);
+    }
+
+protected:
+    kernel_arguments_data get_arguments(typed_primitive_inst<experimental_detectron_generate_proposals_single_image>& instance, int32_t) const override {
+        kernel_arguments_data args;
+        const auto num_inputs = instance.inputs_memory_count();
+        for (size_t i = 0; i < num_inputs; ++i) {
+            args.inputs.push_back(instance.input_memory_ptr(i));
+        }
+
+        args.outputs.push_back(instance.output_memory_ptr());
+        args.inputs.push_back(instance.output_roi_scores_memory());
```

Comment:
Note:  All implementation is assuming that the second output memory is created always and used. 
For performance, I think the second input should be created and used only when it is used. 
If we dont care of performance matter since there is only a ref kernel, it is okay for now. 
However in the future if we need to optimize the memory usage and performance, we need to make it to use second output only when it is needed....

Reply:
agree with you. can we make it in the future as an improvement?

Reply:
done